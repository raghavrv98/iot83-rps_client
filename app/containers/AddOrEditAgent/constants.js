/*
 *
 * AddOrEditAgent constants
 *
 */

export const CREATE_AGENT_REQUEST = "app/AddOrEditAgent/CREATE_AGENT_REQUEST";
export const CREATE_AGENT_SUCCESS = "app/AddOrEditAgent/CREATE_AGENT_SUCCESS";
export const CREATE_AGENT_FAILURE = "app/AddOrEditAgent/CREATE_AGENT_FAILURE";

export const GET_AGENT_DETAILS = "app/AddOrEditAgent/GET_AGENT_DETAILS";
export const GET_AGENT_DETAILS_SUCCESS = "app/AddOrEditAgent/GET_AGENT_DETAILS_SUCCESS";
export const GET_AGENT_DETAILS_FAILURE = "app/AddOrEditAgent/GET_AGENT_DETAILS_FAILURE";
