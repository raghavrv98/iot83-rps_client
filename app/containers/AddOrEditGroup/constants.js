/*
 *
 * AddOrEditGroup constants
 *
 */

export const ON_SUBMIT_REQUEST = "app/AddOrEditGroup/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_FAILURE";

export const GET_GROUP_DETAILS = "app/AddOrEditGroup/GET_GROUP_DETAILS";
export const GET_GROUP_DETAILS_SUCCESS = "app/AddOrEditGroup/GET_GROUP_DETAILS_SUCCESS";
export const GET_GROUP_DETAILS_FAILURE = "app/AddOrEditGroup/GET_GROUP_DETAILS_FAILURE";
