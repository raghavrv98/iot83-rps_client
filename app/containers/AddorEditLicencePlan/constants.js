/*
 *
 * AddorEditLicencePlan constants
 *
 */

export const DEFAULT_ACTION = "app/AddorEditLicencePlan/DEFAULT_ACTION";

export const ON_SUBMIT_REQUEST = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_FAILURE";

export const GET_PLANS_LIST = "app/AddorEditLicencePlan/GET_PLANS_LIST";
export const GET_PLANS_LIST_SUCCESS = "app/AddorEditLicencePlan/GET_PLANS_LIST_SUCCESS";
export const GET_PLANS_LIST_FAILURE = "app/AddorEditLicencePlan/GET_PLANS_LIST_FAILURE";