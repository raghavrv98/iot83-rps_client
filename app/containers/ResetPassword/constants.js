/*
 *
 * ResetPassword constants
 *
 */

export const DEFAULT_ACTION = "app/ResetPassword/DEFAULT_ACTION";

export const RESET_PASSWORD_REQUEST = "app/ResetPassword/RESET_PASSWORD_REQUEST";
export const RESET_PASSWORD_SUCCCESS = "app/ResetPassword/RESET_PASSWORD_SUCCCESS";
export const RESET_PASSWORD_FAILURE = "app/ResetPassword/DEFAULT_ACTION_FAILURE";
