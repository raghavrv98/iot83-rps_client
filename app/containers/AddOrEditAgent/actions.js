/*
 *
 * AddOrEditAgent actions
 *
 */

import { CREATE_AGENT_REQUEST,GET_AGENT_DETAILS } from "./constants";

export function createAgent(data) {
  return {
    type: CREATE_AGENT_REQUEST,
    payload: data,
  };
}
export function getAgentDetails(id) {
  return {
    type: GET_AGENT_DETAILS,
    id
  };
}
