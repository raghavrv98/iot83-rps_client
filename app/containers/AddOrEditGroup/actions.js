/*
 *
 * AddOrEditGroup actions
 *
 */

import { ON_SUBMIT_REQUEST,GET_GROUP_DETAILS } from "./constants";

export function onSubmitHandler(payload,id) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload,
    id
  };
}

export function getGroupDetails(id) {
  return {
    type: GET_GROUP_DETAILS,
    id
  };
}
