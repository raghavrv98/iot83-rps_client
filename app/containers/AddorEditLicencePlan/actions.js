/*
 *
 * AddorEditLicencePlan actions
 *
 */

import { DEFAULT_ACTION, ON_SUBMIT_REQUEST, GET_PLANS_LIST } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function onSubmitHandler(payload) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload
  };
}

export function getPlansList(){
  return{
    type: GET_PLANS_LIST
  }
}