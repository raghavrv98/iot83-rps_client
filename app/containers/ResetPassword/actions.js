/*
 *
 * ResetPassword actions
 *
 */

import { DEFAULT_ACTION, RESET_PASSWORD_REQUEST } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function resetPasswordHandler(payload) {
  return {
    type: RESET_PASSWORD_REQUEST,
    payload
  };
}
