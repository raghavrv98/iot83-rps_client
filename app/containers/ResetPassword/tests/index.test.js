import React from 'react';
import { mount, shallow } from 'enzyme';
// import { enzymeFind } from 'styled-components/test-utils';

import {resetPasswordHandler} from '../actions';
import { ResetPassword, mapDispatchToProps } from '../index';
import {RESET_PASSWORD_REQUEST} from '../constants';

describe('< ResetPassword />', () => {
    it('match snapshots', () => {
        const wrap = shallow(<ResetPassword />)
        expect(wrap).toMatchSnapshot()
      })
  
      it('should test component when mount', () => {
        spyOn(ResetPassword.prototype, 'render').and.callThrough();
        const wrapper = shallow(<ResetPassword />);
        expect(wrapper).toBeDefined();
        expect(ResetPassword.prototype.render).toHaveBeenCalledTimes(1);
      })
  
      it('should test component when recieve props', () => {
        spyOn(ResetPassword.prototype, 'componentWillReceiveProps').and.callThrough();
        const wrapper = shallow(<ResetPassword resetPasswordSuccess={undefined} resetPasswordFailure={undefined} />);
        expect(wrapper).toBeDefined();
        wrapper.setProps({ resetPasswordSuccess:"Reset password success message.", resetPasswordFailure:"this is mock error"});
        expect(ResetPassword.prototype.componentWillReceiveProps).toHaveBeenCalledTimes(1);
      })
      // it("clicking submit button trigger onClick event", async () => {
      //   const onSubmitHandlerSpy = jest.spyOn(ResetPassword.prototype, "resetFormHandler");
      //   const mockLocation = {
      //       search : "mockSearch"
      //   }
      //   const wrapper = (<ResetPassword resetPasswordHandler={resetPasswordHandler} location={mockLocation} />);
      //   let addOrEditForm = wrapper.find("form");
      //   expect(addOrEditForm.length).toBe(1);
      //   addOrEditForm.simulate('submit', { preventDefault() { } });
      //   expect(onSubmitHandlerSpy).toHaveBeenCalledTimes(1);
      // });
      it('Function testing on mapDispatchToProps', () => {
        const dispatch = jest.fn();
        const payload = "demo"
        mapDispatchToProps(dispatch).resetPasswordHandler(payload);
        expect(dispatch.mock.calls[0][0]).
        toEqual({ 
          type: RESET_PASSWORD_REQUEST,
          payload: 'demo',
        })
      });
});