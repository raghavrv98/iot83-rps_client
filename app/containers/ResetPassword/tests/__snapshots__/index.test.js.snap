// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< ResetPassword /> match snapshots 1`] = `
<div
  className="loginBg"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      ResetPassword
    </title>
    <meta
      content="Description of ResetPassword"
      name="description"
    />
  </HelmetWrapper>
  <div
    className="loginFrame"
  >
    <div
      className="loginBranding"
    >
      <div
        className="brandingLogo"
      >
        <img
          onError={[Function]}
          src="undefinedapi/public/static/logo/pmmp.jpeg"
        />
      </div>
      <h6
        className="brandingText"
      >
        <span
          className="brand"
        >
          RAYCHEM
        </span>
        Pipeline Supervisor
      </h6>
    </div>
    <div
      className="loginInfo"
    >
      <div
        className="welcomeText"
      >
        <h4>
          <FormattedMessage
            defaultMessage="WE CONNECT AND PROTECT"
            id="app.common.messages.welcomText"
            values={Object {}}
          >
            <Component />
          </FormattedMessage>
        </h4>
      </div>
      <div
        className="loginForm"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="formRow"
          >
            <input
              className="form-control"
              placeholder="New Password"
              required={true}
              type="password"
            />
          </div>
          <div
            className="formRow"
          >
            <input
              className="form-control"
              placeholder="Confirm Password"
              required={true}
              type="password"
            />
          </div>
          <div
            className="formRow"
          >
            <button
              className="btn-danger"
            >
              <FormattedMessage
                defaultMessage="Reset Password"
                id="app.containers.ResetPassword.resetPassword"
                values={Object {}}
              >
                <Component />
              </FormattedMessage>
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
